core/src/epicli/data/common/ansible/playbooks/roles/ignite/tasks/main.yml

```diff
@@ -0,0 +1,156 @@
+---
+# tasks file for ignite
+- name: add ignite group
+  become: yes
+  become_user: root
+  group:
+    name: ignite
+    state: present
+
+- name: add ignite user
+  become: yes
+  become_user: root
+  user:
+    name: ignite
+    group: ignite
+
+- name: Install Java package
+  package:
+    name: "java-1.8.0-openjdk-headless"
+    state: present
+  when: ansible_os_family == "RedHat"
+
+- name: Install Java package
+  package:
+    name: "openjdk-8-jre-headless"
+    state: present
+  when: ansible_os_family == "Debian"
+
+- name: Set java home variable
+  shell: "which java | xargs readlink -f | xargs dirname | xargs dirname"
+  register: java_home_location
+  failed_when: "java_home_location.rc != 0"
+  changed_when: false
+
+- name: Set JAVA_HOME environment variable
+  lineinfile: 
+    path: /etc/environment 
+    state: present 
+    regexp: '^JAVA_HOME' 
+    line: 'JAVA_HOME={{ java_home_location.stdout }}'
+
+- name: Set Apache Ignite file name to install
+  set_fact:
+    ignite_file_name: "{{ specification.file_name }}"
+  changed_when: false
+
+- name: Check if Ignite in current version exists
+  stat:
+    path: /opt/ignite_{{ specification.version }}/bin/ignite.sh
+  register: ignite_exists
+  changed_when: false
+
+- name: Download Ignite binaries
+  include_role:
+    name: download
+    tasks_from: download_file
+  vars:
+    file_name: "{{ ignite_file_name }}"
+  when: not ignite_exists.stat.exists
+
+- name: Create temp directory
```

Comment:
You can uncompress directly (without temp directory), then move contents from the `apache-ignite-2.7.6-bin` sub-directory (moving should be much faster than copying), see tested example:
```
- name: Create {{ dest_dir }} directory
  file:
    path: "{{ dest_dir }}"
    state: directory
    owner: ignite
    group: ignite
  vars:
    dest_dir: /opt/ignite_{{ specification.version }}

- name: Uncompress {{ ignite_file_name }} to {{ dest_dir }}
  unarchive:
    remote_src: yes
    src: "{{ download_directory }}/{{ ignite_file_name }}"
    dest: "{{ dest_dir }}"
    owner: ignite
    group: ignite
    list_files: yes # to get first level directory name, since unzip doesn't support --strip-components=1
  register: archive_contents
  vars:
    dest_dir: /opt/ignite_{{ specification.version }}

- name: Move contents from {{ src_dir }} to {{ dest_dir }}
  shell: >-
    shopt -s dotglob &&
    mv -f {{ src_dir }}/* {{ dest_dir }} &&
    shopt -u dotglob &&
    rmdir {{ src_dir }}
  args:
    removes: "{{ src_dir }}"
  vars:
    src_dir:  /opt/ignite_{{ specification.version }}/{{ archive_contents.files[0].split('/')[0] }}
    dest_dir: /opt/ignite_{{ specification.version }}
```